x-pack/dockerlogbeat/pipelinemanager/config.go

```diff
@@ -0,0 +1,143 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package pipelinemanager
+
+import (
+	"crypto/sha1"
+	"reflect"
+	"sort"
+
+	"github.com/pkg/errors"
+
+	"github.com/elastic/beats/v7/libbeat/common"
+	"github.com/elastic/beats/v7/libbeat/common/transform/typeconv"
+)
+
+// ContainerOutputConfig has all the options we'll expect from --log-opts
+type ContainerOutputConfig struct {
+	Endpoint        string `struct:"output.elasticsearch.hosts"`
+	User            string `struct:"output.elasticsearch.username"`
+	Password        string `struct:"output.elasticsearch.password"`
+	Index           string `struct:"output.elasticsearch.index"`
+	Pipeline        string `struct:"output.elasticsearch.pipeline"`
+	APIKey          string `struct:"output.elasticsearch.api_key"`
+	Timeout         string `struct:"output.elasticsearch.timeout,omitempty"`
+	BackoffInit     string `struct:"output.elasticsearch.backoff.init,omitempty"`
+	BackoffMax      string `struct:"output.elasticsearch.backoff.max,omitempty"`
+	CloudID         string `struct:"cloud.id"`
+	CloudAuth       string `struct:"cloud.auth"`
+	ProxyURL        string `struct:"output.elasticsearch.proxy_url"`
+	ILMEnabled      bool   `struct:"setup.ilm.enabled"`
+	ILMRollverAlias string `struct:"setup.ilm.rollover_alias"`
+	ILMPatterns     string `struct:"setup.ilm.pattern"`
+	TemplateName    string `struct:"setup.template.name"`
+	TempatePattern  string `struct:"setup.template.pattern"`
+}
+
+// NewCfgFromRaw returns a ContainerOutputConfig based on a raw config we get from the API
+func NewCfgFromRaw(input map[string]string) (ContainerOutputConfig, error) {
+
+	newCfg := ContainerOutputConfig{}
+	endpoint, ok := input["endpoint"]
+	if !ok {
+		return newCfg, errors.New("An endpoint flag is required")
+	}
+	newCfg.Endpoint = endpoint
+
+	var isIndex bool
+
+	newCfg.User, _ = input["user"]
+	newCfg.Password, _ = input["password"]
+	newCfg.Index, isIndex = input["index"]
+	newCfg.Pipeline, _ = input["pipeline"]
+	newCfg.CloudID, _ = input["cloud_id"]
+	newCfg.CloudAuth, _ = input["cloud_auth"]
+	newCfg.ProxyURL, _ = input["proxy_url"]
+	newCfg.APIKey, _ = input["api_key"]
+	newCfg.Timeout, _ = input["timeout"]
+	newCfg.BackoffInit, _ = input["backoff_init"]
+	newCfg.BackoffMax, _ = input["backoff_max"]
```

Comment:
in go a map access for an unknown key returns the "empty" value. No need for `, _`